crates/blockchain-tree/src/externals.rs

```diff
@@ -39,6 +42,22 @@ impl<DB, EVM> TreeExternals<DB, EVM> {
 }
 
 impl<DB: Database, EVM> TreeExternals<DB, EVM> {
+    /// Retrieves parent block total difficulty and returns `true` if the block is pre-merge.
+    pub(crate) fn is_canonical_block_pre_merge(
+        &self,
+        parent_hash: B256,
+    ) -> Result<bool, CanonicalError> {
```

Comment:
Result<bool, _> is an anti-pattern, let's return just Result<()> and roll the `true` case in the `CanonicalError`  variant